Shibboleth integration

lucaguindani opened the pull request.

lucaguindani committed:
Modified login view
Modified user view to hide some informations to the shibboleth users
Managed shibboleth sessions in the zend authentification middleware
Modified nginx configuration & added apache configuration

lucaguindani committed:
Added https urls to proxy file for production

lucaguindani committed:
Modified UserRepository

PowerKiKi commented on server/Application/Middleware/AuthenticationMiddleware.php:
```diff
@@ -53,4 +55,28 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
+     */
+    private function shibboleth(SessionInterface $session): void
+    {
+        if (isset($_SERVER['Shib-Identity-Provider']) && !$session->has('user')) {
```
Je penses que tu devrais utiliser `$request->getServerParams();` à la place de `$_SERVER`. C'est plus propre pour un middleware et te simplifiera la vie pour écrire des unit tests.

PowerKiKi commented on server/Application/Repository/UserRepository.php:
```diff
@@ -101,6 +101,48 @@ public function getOneById(int $id): ?User
+     * @param string $type
+     * @param string $role
+     *
+     * @return null|User
```
Le type de retour est jamais `null`

PowerKiKi commented on server/Application/Repository/UserRepository.php:
```diff
@@ -101,6 +101,48 @@ public function getOneById(int $id): ?User
+     *
+     * @return null|User
+     */
+    public function createShibboleth(string $login, string $mail, string $type = User::TYPE_UNIL, string $role = User::ROLE_STUDENT)
```
Tu devrais probablement typer la méthode ici avec `: User`.

Et probablement que les paramètres par défaut devrait être simplement hardcodé dans la méthode puisque ils ne sont jamais défini par l'appellant.

PowerKiKi reviewed (commented):
Qq commentaires en vrac, mais je te laisses le soin de juger si tu veux merger la PR toi-même malgré tout.

lucaguindani committed:
Modified authentication middleware to use getServerParams instead of $_SERVER
Minor changes in UserRepository createShibboleth function
Removed coverage condition in scrutinizer config

lucaguindani commented:
@PowerKiKi Merci pour cette review, j'ai réalisé les quelques changements. Pour info, j'ai supprimé pour le moment la condition Scrutinizer au niveau du coverage, je reprendrai les tests plus tard quand j'ai plus de temps.
